File: api/src/main/java/org/delivery/api/common/exception/ApiException.java
package org.delivery.common.exception;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.delivery.common.error.ErrorCodeInterface;

/* 코틀린으로 변경
@Getter
public class ApiException extends RuntimeException implements ApiExceptionInterface{

    private final ErrorCodeInterface errorCodeInterface;
    private final String errorDescription;

    public ApiException(ErrorCodeInterface errorCodeInterface) {
        super(errorCodeInterface.getDescription());
        this.errorCodeInterface = errorCodeInterface;
        this.errorDescription = errorCodeInterface.getDescription();
    }

    public ApiException(ErrorCodeInterface errorCodeInterface, String errorDescription) {
        super(errorDescription);
        this.errorCodeInterface = errorCodeInterface;
        this.errorDescription = errorDescription;
    }

    public ApiException(ErrorCodeInterface errorCodeInterface, Throwable throwable) {
        super(throwable);
        this.errorCodeInterface = errorCodeInterface;
        this.errorDescription = errorCodeInterface.getDescription();
    }

    public ApiException(ErrorCodeInterface errorCodeInterface, Throwable throwable, String errorDescription) {
        super(throwable);
        this.errorCodeInterface = errorCodeInterface;
        this.errorDescription = errorDescription;
    }
}
*/

File: common/src/main/kotlin/org/delivery/common/exception/ApiException.kt
package org.delivery.common.exception

import jdk.jfr.Description
import org.delivery.common.error.ErrorCodeInterface

class ApiException : RuntimeException, ApiExceptionInterface {

    override val errorCodeInterface: ErrorCodeInterface?

    override val errorDescription: String?


    constructor(errorCodeInterface: ErrorCodeInterface) : super(errorCodeInterface.getDescription()) {
        this.errorCodeInterface = errorCodeInterface
        this.errorDescription = errorCodeInterface.getDescription()
    }

    constructor(
        errorCodeInterface: ErrorCodeInterface,
        errorDescription: String
    ) : super(errorCodeInterface.getDescription()) {
        this.errorCodeInterface = errorCodeInterface
        this.errorDescription = errorDescription
    }

    constructor(
        errorCodeInterface: ErrorCodeInterface,
        throwable: Throwable
    ) : super(throwable) {
        this.errorCodeInterface = errorCodeInterface
        this.errorDescription = errorCodeInterface.getDescription()
    }

    constructor(
        errorCodeInterface: ErrorCodeInterface,
        errorDescription: String,
        throwable: Throwable
    ) : super(throwable) {
        this.errorCodeInterface = errorCodeInterface
        this.errorDescription = errorDescription
    }




}